extensions/console/internal.m

```diff
@@ -0,0 +1,382 @@
+///
+/// Notes:
+///  * See the `hs.drawing.color` entry in the Dash documentation, or type `help.hs.drawing.color` in the Hammerspoon console to get more information on how to specify a color.
+static int console_outputBackgroundColor(lua_State *L) {
+    NSTextView *output = [MJConsoleWindowController singleton].outputView ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        luaL_checktype(L, 1, LUA_TTABLE) ;
+        [output setBackgroundColor:[[LuaSkin shared] luaObjectAtIndex:1 toClass:"NSColor"]] ;
+    }
+
+    [[LuaSkin shared] pushNSObject:[output backgroundColor]] ;
+    return 1 ;
+}
+
+/// hs.console.inputBackgroundColor([color]) -> color
+/// Function
+/// Get or set the color for the background of the Hammerspoon Console's input field.
+///
+/// Parameters:
+/// * color - an optional table containing color keys as described in `hs.drawing.color`
+///
+/// Returns:
+/// * the current color setting as a table
+///
+/// Notes:
+///  * See the `hs.drawing.color` entry in the Dash documentation, or type `help.hs.drawing.color` in the Hammerspoon console to get more information on how to specify a color.
+static int console_inputBackgroundColor(lua_State *L) {
+    NSTextField *input = [MJConsoleWindowController singleton].inputField ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        luaL_checktype(L, 1, LUA_TTABLE) ;
+        [input setBackgroundColor:[[LuaSkin shared] luaObjectAtIndex:1 toClass:"NSColor"]] ;
+    }
+
+    [[LuaSkin shared] pushNSObject:[input backgroundColor]] ;
+    return 1 ;
+}
+
+/// hs.console.smartInsertDeleteEnabled([flag]) -> bool
+/// Function
+/// Determine whether or not objects copied from the console window insert or delete space around selected words to preserve proper spacing and punctuation.
+///
+/// Parameters:
+///  * flag - an optional boolean value indicating whether or not "smart" space behavior is enabled when copying from the Hammerspoon console.
+///
+/// Returns:
+///  * the current value
+///
+/// Notes:
+///  * this only applies to future copy operations from the Hammerspoon console -- anything already in the clipboard is not affected.
+static int console_smartInsertDeleteEnabled(lua_State *L) {
+    NSTextView *output = [MJConsoleWindowController singleton].outputView ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        [output setSmartInsertDeleteEnabled:(BOOL)lua_toboolean(L, 1)] ;
+    }
+
+    lua_pushboolean(L, [output smartInsertDeleteEnabled]) ;
+    return 1 ;
+}
+
+/// hs.console.getHistory() -> array
+/// Function
+/// Get the Hammerspoon console history as an array.
+///
+/// Parameters:
+///  * None
+///
+/// Returns:
+///  * an array containing the history of commands entered into the Hammerspoon console.
+static int console_getHistory(__unused lua_State *L) {
+    [[LuaSkin shared] checkArgs:LS_TBREAK] ;
+    MJConsoleWindowController *console = [MJConsoleWindowController singleton] ;
+
+    [[LuaSkin shared] pushNSObject:[console history]] ;
```

Comment:
It'll be marginally more efficient to just grab a local LuaSkin pointer, rather than calling through to `shared` multiple times in a function.